samples/boot/mcuboot_recovery_entry/README.rst

```diff
@@ -53,18 +53,14 @@ Programming the S115 SoftDevice
 The SoftDevice is automatically flashed when programming the board.
 No additional steps are needed for programming the SoftDevice when building with MCUboot firmware loader mode enabled.
```

Comment:
Does this text about the SoftDevice make sense to keep in the `mcuboot_recovery_xxxx` samples now that the SoftDevice is programmed automatically for both the mcuboot and simple board variants? I think it can be removed. Any good arguments for keeping it?

If we decide to keep it, at least remove the "S115" from the section title.